lib/self-evict.js

```diff
@@ -131,11 +131,11 @@ SelfEvict.prototype.registerHooks = function registerHooks(hooks) {
     }
 
     if (hooks.preEvict) {
-        this.preEvictHooks[name] = hooks.preEvict;
+        this.preEvictHooks[name] = hooks.preEvict.bind(hooks);
```

Comment:
The other option is to not have two separate maps but one (e.g. `hooks`). However, that would require in the way they are ran. Right now `_runHooks` is passed in an object with all the hooks to run; that have to be a `type` (`preEvict` or `postEvict`). I than have to loop through the full `this.hooks`-object, check if `hook[type]` is available, and invoke it when it is. What would you two prefer?